packages/amplify_datastore/ios/Classes/SwiftAmplifyDataStorePlugin.swift

```diff
@@ -622,9 +610,83 @@ public class SwiftAmplifyDataStorePlugin: NSObject, FlutterPlugin {
         result.append(startNode)
         unresolved.remove(startNode)
     }
-    
+
     // TODO: Remove once all configure is moved to the bridge
     func getPlugin() throws -> AWSDataStorePlugin {
         return try Amplify.DataStore.getPlugin(for: "awsDataStorePlugin") as! AWSDataStorePlugin
     }
+    
+    func createErrorHandler(args: [String: Any])  -> DataStoreErrorHandler {
+        var errorHandler: DataStoreErrorHandler
+        if((args["hasErrorHandler"] as? Bool) == true) {
+            errorHandler = { error in
+                let map : [String:Any] = [
+                    "errorCode" : "DataStoreException",
+                    "errorMesage" : ErrorMessages.defaultFallbackErrorMessage,
+                    "details" : FlutterDataStoreErrorHandler.createSerializedError(error: error)
+                ]
+                self.channel!.invokeMethod("errorHandler", arguments: map)
+            }
+        } else {
+            errorHandler = { error in
+                Amplify.Logging.error(error: error)
+            }
+        }
+        return errorHandler
+    }
+    
+    func createConflictHandler(args: [String: Any])  -> DataStoreConflictHandler {
+        var conflictHandler:DataStoreConflictHandler
+        if((args["hasConflictHandler"] as? Bool) == true) {
+            conflictHandler = { conflictData, onDecision in
+                do {
+                    let modelName = conflictData.remote["__typename"] as! String
+                    let local = conflictData.local as! FlutterSerializedModel
+                    let remote = conflictData.remote as! FlutterSerializedModel
+
+                    let map : [String:Any] = [
+                        "modelName" : modelName,
+                        "local" : try local.toMap(
+                            modelSchemaRegistry: self.modelSchemaRegistry,
+                            customTypeSchemaRegistry: self.customTypeSchemaRegistry,
+                            modelName: modelName),
+                        "remote" : try remote.toMap(
+                            modelSchemaRegistry: self.modelSchemaRegistry,
+                            customTypeSchemaRegistry: self.customTypeSchemaRegistry,
+                            modelName: modelName)
+                    ]
+
+                    self.channel!.invokeMethod("conflictHandler", arguments: map){ result in
+                        do {
+                            let resultMap : [String: Any] = result as! [String: Any]
+                            switch(resultMap["resolutionStrategy"] as! String){
+                                case "APPLY_REMOTE": onDecision(.applyRemote)
+                                case "RETRY_LOCAL": onDecision(.retryLocal)
+                                case "RETRY":
+                                    let modelMap : [String: Any] = resultMap["customModel"] as! [String : Any]
```

Comment:
You force unwrap customModel here but it is nullable in Dart

Reply:
switch with guard 

```
guard let modelMap =  resultMap["customModel"] as? [String : Any]
                                        else {
                                            throw DataStoreError.decodingError("Flutter CustomModel map is invalid", "Check the values that are being passed from Dart.")
                                        }
```